alerting/src/main/kotlin/org/opensearch/alerting/workflow/CompositeWorkflowRunner.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.workflow
+
+import org.apache.logging.log4j.LogManager
+import org.opensearch.alerting.BucketLevelMonitorRunner
+import org.opensearch.alerting.DocumentLevelMonitorRunner
+import org.opensearch.alerting.MonitorMetadataService
+import org.opensearch.alerting.MonitorRunnerExecutionContext
+import org.opensearch.alerting.QueryLevelMonitorRunner
+import org.opensearch.alerting.model.MonitorRunResult
+import org.opensearch.alerting.model.WorkflowRunResult
+import org.opensearch.alerting.util.AlertingException
+import org.opensearch.alerting.util.isDocLevelMonitor
+import org.opensearch.alerting.util.isQueryLevelMonitor
+import org.opensearch.commons.alerting.model.CompositeInput
+import org.opensearch.commons.alerting.model.Delegate
+import org.opensearch.commons.alerting.model.Monitor
+import org.opensearch.commons.alerting.model.Workflow
+import org.opensearch.commons.alerting.util.isBucketLevelMonitor
+import java.time.Instant
+import java.time.LocalDateTime
+import java.util.UUID
+
+object CompositeWorkflowRunner : WorkflowRunner() {
+    private val logger = LogManager.getLogger(javaClass)
+
+    override suspend fun runWorkflow(
+        workflow: Workflow,
+        monitorCtx: MonitorRunnerExecutionContext,
+        periodStart: Instant,
+        periodEnd: Instant,
+        dryRun: Boolean
+    ): WorkflowRunResult {
+        val workflowExecutionStartTime = Instant.now()
+
+        val executionId = workflow.id.plus(LocalDateTime.now()).plus(UUID.randomUUID().toString())
+        var workflowResult = WorkflowRunResult(mutableListOf(), workflowExecutionStartTime, null, executionId)
+        val isTempMonitor = dryRun || workflow.id == Workflow.NO_ID
+
+        logger.debug("Workflow ${workflow.id} in $executionId execution is running")
+        val delegates = (workflow.inputs[0] as CompositeInput).sequence.delegates.sortedBy { it.order }
+        var monitors: List<Monitor>
+
+        try {
+            monitors = monitorCtx.workflowService!!.getMonitorsById(delegates.map { it.monitorId }, delegates.size)
+        } catch (e: Exception) {
+            logger.error("Failed to execute workflow. Error: ${e.message}")
```

Comment:
This should be more descriptive and mention it failed to get the monitors.

Reply:
plz never log e.message

it swallows the entire stacktrace
the format should be `log.error("Message", e)`

Reply:
> plz never log e.message
> 
> it swallows the entire stacktrace the format should be `log.error("Message", e)`

Yeah - I agree. Copy-pasted from documentMonitorRunner. Makes sense